src/main/java/org/opennars/storage/Memory.java

```diff
@@ -127,11 +127,13 @@ public void reset() {
         synchronized (concepts) {
             concepts.clear();
         }
-        novelTasks.clear();
         synchronized (newTasks) {
             newTasks.clear();
+            novelTasks.clear();
```

Comment:
Syncronize on noveltasks!

Reply:
No, as novelTasks is used when newTasks lock is already held (as they are used in interaction) wherever it is used, so it would be unnecessary locking.

Reply:
Doesn't matter - the semantics of synchronized (newTasks) is to lock for newTasks(only) - you can rewrite the mutex newTasks to tasksMutex

member:
boolean tasksMutex;

// usage/locking for newTasks and noveltasks
synchronized (tasksMutex) {
   ...
}